### DISCUSSION TITLE
剪贴板需要优化界面和操作逻辑，现在用的有点难受

### DISCUSSION THREAD

**Discussion by erick-zhang-1** [linuxdeepin/developer-center#3975] [Features Request & Ideas | 特性请求 & 头脑风暴]
deepin的功能基本上都好用，就是剪贴板用着很别扭。
为什么要设计的那么大，还不能改位置？
为什么不能用按键选中记录和直接粘贴？
为什么不过滤一样的内容？

# 使用问题
1. 剪贴板ui会占用屏幕的一大片空间，有些时候会影响到使用，并且每条记录占用的位置也大
2. 剪贴板不能使用按键复制粘贴，比如按enter没反应，我平时更多还是使用键盘
3. 从终端复制文本多次，剪贴板里就会显示多次同个内容

# 建议
1. 可以把剪贴板改成一个悬浮的窗口，可以更改位置，大小和固定位置
2. 让剪贴板监听按键事件，支持按上下键选择记录，按enter将记录移到第一位，按enter+shift直接粘贴
3. 过滤剪贴板复制的内容，不保存一样的记录，或者弄一个合计

**Comment by fslong520** [comment]
个人觉得剪贴板可以不那么炫酷，但一定要高效，现在的剪贴板一点都不高效。
另外，剪贴板还有个问题：
- 存储在剪贴板里面的数据的权限会有问题，有些app会偶现不能访问剪贴板的问题，比如**微信、edge、libreoffice**。

比如复制a后想粘贴到微信中，结果剪贴板上面有a，而且是第一位，但粘贴不到微信中，此时如过做以下操作就可以粘贴到微信中了：
1. 剪贴板中双击一下另外的内容b变成当前复制的b；
2. 重新打开剪贴板；
3. 再双击一下a，变成当前复制的a，此时就可以粘贴到微信中了。

如果不做1、2操作，只双击剪贴板中的a依然是不行的，感觉跟权限有关。

**Comment by xzl01** [comment]
cc @meiyixiang

**Comment by a137933** [comment]
以前我也提过一些问题https://github.com/linuxdeepin/developer-center/issues/2639

**Comment by meiyixiang** [comment]
谢谢反馈，这个问题我先记录下来。在23release发布前暂时不会去做功能类型的优化，需求池中我已经记录下来，待我对23后期版本做规划时会考虑此否能该如何去优化和提升效率，后期明确的计划出来后会进行公示。

**Comment by debuggerx01** [comment]
我的enter键选择的PR已经挂在那儿很久了……每次更新我都要自己patch以后编译自用版本。。
[feat: select history item by press 'Enter' key. #126](https://github.com/linuxdeepin/dde-clipboard/pull/126)

**Reply by BLumia** [reply]
@18202781743 看下

**Comment by pmzgit** [comment]
可以参考copyQ的功能和交互：
* 一个快捷键显示剪切版窗口：建议别用ctrl+alt+v作为默认（跟idea冲突）,像这种默认，应该为win+v，当然不重要
* 显示窗口后，按任意键，就是搜索功能，比如，直接按he两个键，就是全文搜索包含he的剪切条目，当直接输入数字时，应该直接导航到数字所在的剪切条目
* 上下键：移动选择要复制的剪贴条目
* enter 键=粘贴，如果用户光标在输入状态(比如文本编辑器里，浏览器导航栏里，或者用户在前两中情况下全选了一段文字，enter键，就直接将选中文本替换为选中的切贴条目）直接粘贴。如果光标不是输入状态，那enter键，就将选中的条目作为第一条，等待用户粘贴
* **最重要的一点，复制粘贴的快捷键，还是要符合win,mac，linux的习惯，统一就是ctrl+insert/ctrl+c=复制，shift+insert/ctrl+v=粘贴，也就是emacs默认的映射，不要搞什么乱其八糟的映射**

你看整个交互，很简单，最快，两次按键，就能完成粘贴，1调出显示窗口，2enter粘贴。这就是80%的使用场景，至于鼠标用户(普通用户)，点击按钮，滚轮选择，点击选中，ctrl+v粘贴(输入模式下，甚至不用ctrl+v)，也是很简单

不知道深度的产品经理，最开始做设计的时候，怎么想的，这个东西有那么难吗？自从知道深度支持剪切版历史，用了一下，果断放弃，实在是差的不行

**Reply by BLumia** [reply]
参见 @meiyixiang https://github.com/orgs/linuxdeepin/discussions/3975#discussioncomment-5504810 的答复。正式版 release 后应该才会去做交互上的优化。

**Comment by BLumia** [comment]
更新，关于部分涉及到的需求可以到下述 Issue 跟进：

- 打开剪切板管理器的默认快捷键调整到 <kbd>Meta+V</kbd>：https://github.com/linuxdeepin/developer-center/issues/4597
- 内容展示优化：https://github.com/linuxdeepin/developer-center/issues/4599
- 当存在输入焦点时，选中即帮助填充：https://github.com/linuxdeepin/developer-center/issues/4600
